lib/services/oauth/oauth_server.go

```diff
@@ -0,0 +1,68 @@
+package oauth
+
+import (
+	"context"
+	"os"
+
+	"github.com/markgravity/golang-ic/helpers/log"
+
+	"github.com/go-oauth2/oauth2/v4/errors"
+	"github.com/go-oauth2/oauth2/v4/manage"
+	"github.com/go-oauth2/oauth2/v4/models"
+	"github.com/go-oauth2/oauth2/v4/server"
+	"github.com/go-oauth2/oauth2/v4/store"
+	pg "github.com/vgarvardt/go-oauth2-pg/v4"
+)
+
+var oauthServer *server.Server
+var clientStore *pg.ClientStore
+
+func SetUpOAuthServer() error {
+	manager := manage.NewDefaultManager()
+	// token store
+	manager.MustTokenStorage(store.NewMemoryTokenStore())
+
+	clientStore := store.NewClientStore()
+
+	err := clientStore.Set(os.Getenv("CLIENT_ID"), &models.Client{
+		ID:     os.Getenv("CLIENT_ID"),
+		Secret: os.Getenv("CLIENT_SECRET"),
+		Domain: os.Getenv("DOMAIN"),
+	})
```

Comment:
```suggestion
  client := models.Client{
		ID:     os.Getenv("CLIENT_ID"),
		Secret: os.Getenv("CLIENT_SECRET"),
		Domain: os.Getenv("DOMAIN"),
	}
	
	err := clientStore.Set(client.ID, &client)
```

Not sure if it works as I don't know if the `client` object exposes the `ID` attribute.

In can that works:
- calling the `os.Getenv("CLIENT_ID")` only once instead of twice
- The next line `if err != nil` is closer to the method name, so it's a bit more readable